### PR TITLE
Fix defect that the state of "Run Automatically" option is not maintained

### PR DESCRIPTION
**Pull request by ke-yu**
This PR tries to fix the defect that [the state of "Run Automatically" option is not maintained] (http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6212) when opening a new file or clear the current workspace. 

The root cause is whenever a home workspace is created (through opening a new .dyn file) or cleared, ``Homeworkspace.DynamicRunEnabled`` is never set to the current value, and ``HomeWorkspaceModel`` constructor always calls ``ResetEngine()`` to run the graph.

@Benglin  please help to review the change.

### CODE REVIEW

**Comment by sharadkjaiswal** [comment]
@aparajit-pratap you have worked on related area, please take a look.

**Comment by aparajit-pratap** [comment]
This relates to: https://github.com/DynamoDS/Dynamo/pull/3553

This behaviour was changed after the recent refactoring and I was seeking UX input to clarify the new expectations.

**Comment by ke-yu** [comment]
Thanks, @aparajit-pratap , verified this test case passes now, remove Failure category.

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -889,9 +889,12 @@ private void WorkspaceAdded(WorkspaceModel item)
             var newVm = new WorkspaceViewModel(item, this);
             if (item is HomeWorkspaceModel)
             {
+                (item as HomeWorkspaceModel).DynamicRunEnabled = DynamicRunEnabled;
```
Comment (Benglin) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: I am not sure if this is a no-op, `DynamicRunEnabled` (the right hand side) is as follow:

```csharp
public bool DynamicRunEnabled
{
    get
    {
        return HomeSpace.DynamicRunEnabled;
    }
}

public HomeWorkspaceModel HomeSpace
{
    get
    {
        return model.Workspaces.OfType<HomeWorkspaceModel>().FirstOrDefault();
    }
}
```

When we get to `WorkspaceAdded(WorkspaceModel item)` method, the `model.Workspaces` would have been updated with the incoming `item` here. In another word, `item` here is the same as `HomeSpace`, so the above assignment may not do anything. Am I right?

**Comment by Benglin** [comment]
Apart from the doubt above, the rest LGTM!

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs (hunk 1 above)
Comment (ke-yu) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: Although ``item`` is not the same as ``HomeSpace`` here, but your point is valid. Here we take the assumption that newly added home workspace model is *after* the current home workspace model, this assumption is brittle. I'll do the change. Thanks!

**Comment by Benglin** [comment]
Thanks, the fix LGTM!

**Pull request merged**